File: Daily Challenge/Python/K - Digit Integers - Xth Smallest 20.10.2025.py
N = int(input())
c = []
arr = input().split()
k, x = map(int, input().split())
for i in arr:
    if int(i) < 0 and len(i) - 1 == k:
        c.append(int(i))
    elif int(i) > 0 and len(i) == k:
        c.append(int(i))
c.sort()
if len(c) < x:
    print(-1)
else:
    print(c[x-1])

File: Daily Challenge/Java/K - Digit Integers - Xth Smallest 20.10.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        List<String> arr = new ArrayList<>();
        for (int i = 0; i < N; ++i) arr.add(sc.next());
        int k = sc.nextInt();
        int x = sc.nextInt();
        List<Integer> c = new ArrayList<>();
        for (String str : arr) {
            int num = Integer.parseInt(str);
            int len = str.length();
            if ((num < 0 && len - 1 == k) || (num > 0 && len == k))
                c.add(num);
        }
        Collections.sort(c);
        if (c.size() < x)
            System.out.println(-1);
        else
            System.out.println(c.get(x-1));
    }
}